test/DictoVec.jl

```diff
@@ -48,7 +48,7 @@ end
     @test show2string(dv) == "DictoVec{Symbol}()"
 
     # add an element
-    @test_throws MethodError (dv[1] = "xx") # incompatible value
+    @test_throws Exception (dv[1] = "xx") # incompatible value
```

Comment:
This test has two issues: referring to a non-existing element by position (throws `BoundsError`) and assigning incompatible value (`MethodError`).
It seems in 1.11 the priority has changed to the first check.
`Exception` is too broad; to specifically check for the incompatible value we probably need to change the test to something like

```suggestion
    @test_throws MethodError (dv["A"] = "xx") # incompatible value
```